src/main/java/org/example/pageobject/filteringandsearchingpages/GamingKeyboardsCategoryPage.java

```diff
@@ -0,0 +1,208 @@
+        JavascriptExecutor js = (JavascriptExecutor) webDriver;
+        js.executeScript("arguments[0].click();", brandsExpander);
+
+        WebElement brandNameCheckbox = new WebDriverWait(webDriver, Duration.ofSeconds(6))
+                .until(ExpectedConditions.presenceOfElementLocated(By.xpath("//span[text() = '" + brandName + "']")));
+        brandNameCheckbox.click();
+        return this;
+    }
+
+    public GamingKeyboardsCategoryPage setPriceRange(float minPrice, float maxPrice) {
+        waitForElementVisibility(minPriceInput);
+        minPriceInput.sendKeys(String.valueOf(minPrice));
+        maxPriceInput.sendKeys(String.valueOf(maxPrice));
+        waitForElementVisibility(submitPriceRangeBtn);
+        submitPriceRangeBtn.click();
+        return this;
+    }
+
+    public GamingKeyboardsCategoryPage sortProductsByPriceLowToHigh() {
+        waitForElementVisibility(sortingDropdownList);
+        sortingDropdownList.click();
+        lowToHighDropdownSelection.click();
+        return this;
+    }
+
+    public boolean verifyEveryTitleContainsBrandName(String brandName) {
+        boolean everyTitleContainsInputWord;
+
+        while (true) {
+            waitForElementPresence(titleXpath);
+
+            everyTitleContainsInputWord = titlesList
+                    .stream()
+                    .map(WebElement::getText)
+                    .map(String::toLowerCase)
+                    .allMatch(e -> e.contains(brandName.toLowerCase()));
+
+            if (!everyTitleContainsInputWord) {
+                break;
+            }
+
+            try {
+                waitForElementVisibility(paginationNextBtn);
+            } catch (Exception e) {
+                break;
+            }
+
+            if (paginationNextBtn.isDisplayed() && !(paginationNextBtn.getAttribute("aria-disabled") == null)) {
+                break;
+            }
+
+            paginationNextBtn.click();
+        }
+        return everyTitleContainsInputWord;
+    }
+
+    public boolean verifyPricesAreInChosenRange(float minPrice, float maxPrice) {
+        boolean arePricesInChosenRange;
+
+        while (true) {
+            waitForElementPresence(itemContainersXpath);
+
+            arePricesInChosenRange = itemContainers
+                    .stream()
+                    .filter(e -> e.findElements(By.xpath(sponsoredSpanXpath)).isEmpty()
+                            && !e.findElements(By.xpath(biggerPriceTextXpath)).isEmpty())
+                    .map(e -> Float.parseFloat(e.findElement(By.xpath(biggerPriceTextXpath))
+                            .getAttribute("textContent").replace("$", "")))
+                    .allMatch(price -> price >= minPrice && price <= maxPrice);
+
+            if (!arePricesInChosenRange) {
+                break;
+            }
+
+            try {
+                waitForElementVisibility(paginationNextBtn);
+            } catch (Exception e) {
+                break;
```

Comment:
Also it is better to add some log. At least System.err.println("Info what went wrong")